cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/UpdateController.java

```diff
@@ -80,42 +81,61 @@ public Integer call() throws Exception {
         @Option(names = { "-f", "--force" }, required = false, description = "Force deletion without confirmation.")
         Boolean force;
 
+        private Map<String, ServiceBean> readServicesFromYAMLFile() throws IOException {
+            SpecBean specBean = YamlMapper.loadBean(yamlCommandOptions.getYamlFilePath(), SpecBean.class);
+            return specBean.getServices();
+        }
+
         @Override
         public Integer call() throws Exception {
+            Map<String, ServiceBean> services = readServicesFromYAMLFile();
 
-            if (force != null) {
-                doRemoveServiceInstance(yamlCommandOptions);
-            } else {
+            // in case the --force flag is not specified, we shall prompt the user to prevent the removal of services
+            // as that might remove valuable data a swell
+            if (force == null) {
                 if (System.console() == null) {
                     Log.error("--force/-f not supplied and not running in terminal, aborting");
                     return 2;
                 }
 
-                System.out.println("Really delete the services y/n?");
+                System.out.print("Requested removal of the following services: ");
+                for (String serviceName : services.keySet()) {
+                    System.out.print("\"" + serviceName + "\" ");
+                }
+                System.out.println();
+
+                System.out.print("Are you sure you want to permanently delete those? [y|N] ");
+
                 Scanner scanner = new Scanner(System.in);
                 String input = scanner.nextLine();
                 scanner.close();
-                if (input.equals("y") || input.equals("yes")) {
-                    doRemoveServiceInstance(yamlCommandOptions);
-                } else {
-                    System.out.println("Delete cancelled");
-                    
+
+                // turn into lower case, we would accept an upper case response, too
+                input = input.toLowerCase();
+
+                if (!(input.equals("y") || input.equals("yes"))) {
+                    System.out.println("Cancelled by user");
                     return 1;
                 }
             }
+
+            // in case the service doesn't exist on the server, it'll throw that exception
+            // we can recognize that and log it nicely
+            try {
+                doRemoveServiceInstance(services);
+            } catch (IllegalArgumentException e) {
+                Log.error(e);
+            }
```

Comment:
Wouldn't it be better to log a warning and continue with the deletion of subsequent services? This would require to move the error handling into the doRemoveServiceInstance method.

Reply:
I didn't want to touch the logic of the deletion, I just noticed that issue. We can do that, but I'd first like to ask @yenle2501 and @alexanderKoltman about their opinion.

Reply:
I think the idea of felix is good.